src/backend/distributed/planner/function_call_delegation.c

```diff
@@ -300,7 +417,7 @@ TryToDelegateFunctionCall(DistributedPlanningContext *planContext)
 	/* return if we could not find a placement */
 	if (placement == NULL)
 	{
-		return false;
+		return NULL;
 	}
 
 	WorkerNode *workerNode = FindWorkerNode(placement->nodeName, placement->nodePort);
```

Comment:
below we have the following check: `if (workerNode->groupId == GetLocalGroupId())` 

I think we should call CheckDelegatedFunctionExecution in that scenario to make sure we do not relieve restrictions just because the data is local.

Reply:
It was there in the original PR
![image](https://user-images.githubusercontent.com/44680808/149841060-e0be2628-ef53-46b9-9c3e-506d7aee1c5c.png)